File: Week 1 Assigment Using C/6_Check_sorted_rotated.c

#include <stdio.h>
int isSortedAndRotated(int arr[], int n) {
    int count = 0;
    for(int i = 0; i < n; i++)
        if(arr[i] > arr[(i+1)%n]) count++;
    return (count <= 1);
}
int main() {
    int arr[] = {3, 4, 5, 1, 2}, n = 5;
    printf("%s", isSortedAndRotated(arr, n) ? "Yes" : "No");
    return 0;
}

File: Week 1 Assignment using Python/6_Check_sorted_rotated.py

def is_sorted_and_rotated(arr):
    count = 0
    for i in range(len(arr)):
        if arr[i] > arr[(i + 1) % len(arr)]:
            count += 1
    return count <= 1

arr = [3, 4, 5, 1, 2]
print("Yes" if is_sorted_and_rotated(arr) else "No")
